docs/introduction.md

```diff
@@ -67,83 +96,207 @@ Flow is based upon the following set of rules:
+A guaranteed collection is broadcast by the guarantors to all consensus nodes.
+
+To vote to append a transaction to a collection, a collection node must verify that:
+1. The collection node has received all the relevant transactions
+2. The transaction is well-formed
+3. Appending the transaction does not result in duplicate transactions
+4. There are no common transactions between the current collection under construction and any other collection that the cluster of the collection node already guaranteed
+
+Collection node must store all the transactions of all guaranteed collections, and must produce relevant transaction when requested by the execution nodes. If they fail to do so, they will be slashed, along with the rest of the cluster.
+
+### Consensus Nodes
+
+The consensus nodes work with transaction batches - [_collections_](#collection-nodes), submitted by the collection nodes.
 They form blocks from collections, are in charge of sealing those blocks, and adjudicate slashing requests from other nodes (for example, claims that an execution node has produced incorrect outputs.)
-A block contains collection hashes, and a source of randomness which is used to shuffle the transactions before computing them.
-consensus nodes don't directly compute the transaction order for a block, but they implicitly determine it by specifying all the given inputs to a deterministic algorithm that computes the order.
 
 Since the responsibility to maintain a large state is delegated to specialized nodes, hardware requirements for consensus nodes remain moderate even for high-throughput blockchains.
 As opposed to other node roles, consensus nodes deal with subjective problems, where there is no single correct answer.
 Instead, one answer must be selected through mutual agreement, which is why it is critical for consensus nodes to be numerous and decentralized, and less so for other nodes.
 This design increases decentralization by allowing for higher levels of participation in consensus by individuals with suitable consumer hardware on home internet connections.
 
 When a consensus node receives a guaranteed collection of transactions, it has to run its consensus algorithm to reach an agreement with other nodes over the set of collections to be included in the next block.
-A block of the ordered collection that has undergone the complete consensus algorithm is called a _finalized block_.
+A block of the ordered collections that has undergone the complete consensus algorithm is called a **finalized block**.
 A block specifies the included collections as well as the other inputs (randomness seed, etc.) which are required to execute the computation.
 It is worth noting that a block in Flow does not include the resulting execution state of the block execution.
 
 In order for consensus nodes to seal blocks, they must commit to the execution result of a block after it is executed and verified.
```

Comment:
I think we should explain what sealing means somewhere in this paragraph.

_A block can be sealed after it has been executed by the execution nodes and the execution result has been verified by a sufficient number of verification nodes. When consensus nodes received enough result approvals to agree on the execution result of a block, they include a seal attesting to the execution state after the block execution into the chain, at which point the block is considered sealed._